src/androidTest/kotlin/com/criptext/mail/scenes/mailbox/data/EmptyTrashWorkerTest.kt

```diff
@@ -0,0 +1,106 @@
+package com.criptext.mail.scenes.mailbox.data
+
+import android.support.test.rule.ActivityTestRule
+import android.support.test.runner.AndroidJUnit4
+import com.criptext.mail.androidtest.TestActivity
+import com.criptext.mail.androidtest.TestDatabase
+import com.criptext.mail.api.HttpClient
+import com.criptext.mail.api.models.EmailMetadata
+import com.criptext.mail.db.DeliveryTypes
+import com.criptext.mail.db.MailboxLocalDB
+import com.criptext.mail.db.models.ActiveAccount
+import com.criptext.mail.db.models.Contact
+import com.criptext.mail.db.models.Label
+import com.criptext.mail.utils.MockedResponse
+import com.criptext.mail.utils.enqueueResponses
+import io.mockk.mockk
+import okhttp3.mockwebserver.MockWebServer
+import org.amshove.kluent.shouldBe
+import org.junit.After
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4::class)
+class EmptyTrashWorkerTest{
+
+    @get:Rule
+    val mActivityRule = ActivityTestRule(TestActivity::class.java)
+
+    private lateinit var db: TestDatabase
+    private lateinit var mailboxLocalDB: MailboxLocalDB
+    private val activeAccount = ActiveAccount(name = "Tester", recipientId = "tester",
+            deviceId = 1, jwt = "__JWTOKEN__", signature = "")
+    private lateinit var httpClient: HttpClient
+    private lateinit var mockWebServer: MockWebServer
+
+    @Before
+    fun setup() {
+        // mock http requests
+        mockWebServer = MockWebServer()
+        mockWebServer.start()
+        val mockWebServerUrl = mockWebServer.url("/mock").toString()
+        httpClient = HttpClient.Default(authScheme = HttpClient.AuthScheme.jwt,
+                baseUrl = mockWebServerUrl, connectionTimeout = 1000L, readTimeout = 1000L)
+        db = TestDatabase.getInstance(mActivityRule.activity)
+        db.resetDao().deleteAllData(1)
+        db.labelDao().insertAll(Label.DefaultItems().toList())
+        mailboxLocalDB = MailboxLocalDB.Default(db)
+
+        insertEmailsNeededForTests()
+    }
+
+    private fun insertEmailsNeededForTests() {
```

Comment:
Isn't there a way to generalize this and avoid writing the same function in every test file?